### PR TITLE
klog-rs: init at 0.0.3

### PR DESCRIPTION
**Pull request by tobifroe**
This adds [klog-rs](https://github.com/tobifroe/klog), a small utility simplifying the process of working with Kubernetes logs.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/kl/klog-rs/package.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    hash = "sha256-HEGxg277483ZZpXKFIBTAITKie/iBfbz9KmtqzlGC3E=";
+  };
+  cargoHash = "sha256-mOFzlq5AEks9vwjk2FxTGLyw6RkI44kml2t8r1UsdOk=";
+  checkFlags = [
+    # this integration test depents on a running kubernetes cluster
+    "--skip=k8s::tests::test_get_pod_list"
+  ];
+
+  meta = {
+    description = "klog is a tool that allows you to tail logs of multiple Kubernetes pods simultaneously.";
```
Comment (SigmaSquadron) [inline pkgs/by-name/kl/klog-rs/package.nix:24]:
```suggestion
    description = "Tool to tail logs of multiple Kubernetes pods simultaneously";
```
See the [Meta Attributes chapter](https://nixos.org/manual/nixpkgs/unstable/#var-meta-description) of the Nixpkgs Manual.

**Review by SigmaSquadron — COMMENTED** [review]
Welcome to Nixpkgs! I'm sorry this went unreviewed; the diff looks very high-quality.

File: pkgs/by-name/kl/klog-rs/package.nix (hunk 1 above)
Comment (tobifroe) [reply pkgs/by-name/kl/klog-rs/package.nix]: Adapted the description, thank you!

File: pkgs/by-name/kl/klog-rs/package.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  };
+  cargoHash = "sha256-mOFzlq5AEks9vwjk2FxTGLyw6RkI44kml2t8r1UsdOk=";
+  checkFlags = [
+    # this integration test depents on a running kubernetes cluster
+    "--skip=k8s::tests::test_get_pod_list"
+  ];
+
+  meta = {
+    description = "Tool to tail logs of multiple Kubernetes pods simultaneously";
+    homepage = "https://github.com/tobifroe/klog";
```
Comment (h7x4) [inline pkgs/by-name/kl/klog-rs/package.nix:25]: Not a blocker :)

```suggestion
    homepage = "https://github.com/tobifroe/klog";
    changelog = "https://github.com/tobifroe/klog/releases/tag/${version}";
```

**Review by h7x4 — COMMENTED** [review]
Welcome to nixpkgs :tada:

I havent' tested the kubernetes part, but everything compiles correctly. The diff looks good.

Could you fix up your commits? Optimally it should be two commits like this. Make sure to order the maintainer list change before the package commit.

```
maintainers: add tobifroe
klog-rs: init at 0.0.3
```

File: pkgs/by-name/kl/klog-rs/package.nix (hunk 2 above)
Comment (tobifroe) [reply pkgs/by-name/kl/klog-rs/package.nix:25]: Added that, thank you! :)

**Comment by tobifroe** [comment]
Thank you for reviewing my PR, I just reordered and squashed my commits. Is there anything else for me to do?

**Comment by h7x4** [comment]
Everything looks good. Verified the github id and the license upstream.

Could you rebase your commits on main? ofborg is throwing a fit, but I don't think it's a problem with your code

**Review by h7x4 — APPROVED** [review]

**Comment by tobifroe** [comment]
Thank you @h7x4, just rebased on main

**Comment by h7x4** [comment]
It looks like darwin needs some extra libraries for the program to compile. You could either mark it as broken on darwin, or try to fix it up. I'm not too familiar with it, but there should be plenty of people to ping if you need any help :)

**Comment by h7x4** [comment]
I think you might've forgotten to stage the changes or something. The last forcepush didn't seem to have changed anything.

**Comment by Aleksanaa** [comment]
Just wait for this staging-next till tomorrow. Completing `darwin.apple_sdk.frameworks.xxx` won't be necessary after that

**Pull request merged**